modules/account-lib/src/mpc/tss.ts

```diff
@@ -0,0 +1,304 @@
+/**
+ * Module provides functions for MPC using threshold signature scheme (TSS). It contains
+ * functions for key generation and message signing with EdDSA.
+ *
+ *
+ * ======================
+ * EdDSA Key Generation
+ * ======================
+ * 1. Each signer generates their own key share, which involves a private u-share and a public y-share.
+ * 2. Signers distribute their y-share to other signers.
+ * 3. After exhanging y-shares the next phase is to combine key shares. Each signer combines their u-share
+ *    with the y-shares received from other signers in order to generate a p-share for themselves. We
```

Comment:
@johnoliverdriscoll Q for here, is there a specific format that's expected when we save data into the hsm? even though here I'm saying that I'm generating a u-share, I'm not really generating that as a single string value. instead, it's the whole json from

```
interface PlayerKeyShare {
  i: string;
  y: string;
  u: string;
  prefix: string;
}
```

while an x-share is the whole collection from: 

```
interface PlayerSignShare {
  i: string;
  y: string;
  x: string;
  r: string;
  R: string;
}
```